Sources/mParticle-Appboy/MPKitAppboy.m

```diff
@@ -425,37 +426,105 @@ - (MPKitExecStatus *)routeCommerceEvent:(MPCommerceEvent *)commerceEvent {
+                
+                [appboyInstance logPurchase:product.sku
+                                   currency:currency
+                                      price:[product.price doubleValue]
+                                   quantity:[product.quantity integerValue]
+                                 properties:properties];
+                
+                [execStatus incrementForwardCount];
+            }
         }
     } else {
-        NSArray *expandedInstructions = [commerceEvent expandedInstructions];
-        
-        for (MPCommerceEventInstruction *commerceEventInstruction in expandedInstructions) {
-            [self logBaseEvent:commerceEventInstruction.event];
+        if (_configuration[bundleProductsWithCommerceEvents] && [_configuration[bundleProductsWithCommerceEvents] boolValue]) {
+            NSDictionary *transformedEventInfo = [commerceEvent.customAttributes transformValuesToString];
+            
+            NSMutableDictionary *eventInfo = [[NSMutableDictionary alloc] initWithCapacity:commerceEvent.customAttributes.count];
+            [transformedEventInfo enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull key, id _Nonnull obj, BOOL * _Nonnull stop) {
+                NSString *strippedKey = [self stripCharacter:@"$" fromString:key];
+                eventInfo[strippedKey] = obj;
+            }];
+            
+            if (self->_enableTypeDetection) {
+                eventInfo = [[self simplifiedDictionary:eventInfo] mutableCopy];
+            }
+            
+            NSArray *productArray = [self getProductListParameters:commerceEvent.products];
+            if (productArray.count > 0) {
+                [eventInfo setValue:productArray forKey:productKey];
+            }
+            NSArray *promotionArray = [self getPromotionListParameters:commerceEvent.promotionContainer.promotions];
+            if (promotionArray.count > 0) {
+                [eventInfo setValue:promotionArray forKey:promotionKey];
+            }
+            NSArray *impressionArray = [self getImpressionListParameters:commerceEvent.impressions];
+            if (impressionArray.count > 0) {
+                [eventInfo setValue:impressionArray forKey:impressionKey];
+            }
+            
+            NSString *eventName = [NSString stringWithFormat:@"eCommerce - %@", [self eventNameForAction:commerceEvent.action]];
+            if ([eventName isEqualToString:@"eCommerce - unknown"]) {
+                if (commerceEvent.impressions != nil) {
+                    eventName = @"eCommerce - impression";
+                } else if (commerceEvent.promotionContainer.action) {
+                    eventName = [NSString stringWithFormat:@"eCommerce - %@", [self eventNameForPromotionAction:commerceEvent.promotionContainer.action]];
+                }
+            }
+            
+            // Appboy expects that the properties are non empty when present.
+            if (eventInfo && eventInfo.count > 0) {
```

Comment:
```suggestion
            if (eventInfo.count > 0) {
```

Similar suggestion as in the previous places. If `eventInfo` is nil, then count will return 0, so the first check is unnecessary.